### ISSUE TITLE
[Core][Mechanic] Leeway / Melee attack range.

### ISSUE THREAD

**Issue by Sattva-108** [Atlantiss/NetherwingBugtracker#8217]
# **Description**
Leeway doesn't work properly (or doesn't work at all for players) on PTR of Netherwing. 

# **Current behaviour**
**World mobs** have attack range around `10 yards` or even more (should be 8) while both player and mob is moving.
**Player** doesn't seem to have any leeway working for both yellow and white dmg or at least it's not same as mob attack range. 
With increased movespeed you can easily see that mobs hits you from "Narnia" and `you can't attack him.`

# **Expected behaviour**
It's simple. When both player and mob are moving, both should be able to hit each other from 8 yards range. 
If either not moving it should be 5 yards range (it's working)

Leeway mechanic is very well explained in this [small video](https://www.youtube.com/watch?v=S3GH5cWPFOE&t=1s)

# **Server Revision**
Netehrwing PTR TBC (`.server info` not working)

# **Extra materials**
Enemy target [attack bar addon](https://mega.nz/file/uUUFlQ7T#p1GF3amvmNVq7o0yFfWtbTMA47rZ2cuIwSn3LS7q0_Y) i use.
**I have two bug-evidence video** recorded just now in Netherwing PTR. 
**Basically what i do is:** Run from mob and spamming my **Mangle (Cat)** macro with `/startattack` combined. 

https://user-images.githubusercontent.com/74269253/226543191-2faf1194-5878-4345-b65d-0b0aa695d12d.mp4

https://user-images.githubusercontent.com/74269253/226543199-78ad23ed-2af4-4bb8-b461-a567755b04e5.mp4

**Comment by Wolffenstein** [comment]
Melee range is usually 7.66yd but not always, as it depends on both entities' combat reach (in which case with some large mobs the combat reach may exceed that).

The issue is that the range check for player melee swings didn't include leeway whereas creatures' did, so the mob could attack you while you couldn't attack back. The correct scenario in that case is that both you and creature should be able to attack each other. 

Will be fixed in rev 3726.
